magicctapipe/scripts/lst1_magic/lst_m1_m2_cloud_correction.py

```diff
@@ -232,37 +458,85 @@ def process_telescope_data(input_file, config, tel_id, camgeom, focal_eff):
             raise ValueError("Error: 'inds_img' list is empty!")
         index_img = inds_img[0]
         image = dl1_images["image"][index_img]
-        cleanmask = dl1_images["image_mask"][index_img]
+        clean_mask = dl1_images["image_mask"][index_img]
         peak_time = dl1_images["peak_time"][index_img]
         image /= trans_pixels
-        corr_image = image.copy()
 
-        clean_image = corr_image[cleanmask]
-        clean_camgeom = camgeom[cleanmask]
+        # additional cleaning
+        if config["cloud_correction"]["use_additional_cleaning"]:
+            clean_mask = np.ones_like(
+                image, dtype=bool
+            )  # Assuming full mask if not defined
+            if tel_ids["LST-1"] == tel_id:
+                clean = tailcuts_clean(
+                    camgeom,
+                    image,
+                    boundary_thresh=modified_boundary_thresh_lst,
+                    picture_thresh=modified_picture_thresh_lst,
+                    min_number_picture_neighbors=cleaning_level_lst[
+                        "min_number_picture_neighbors"
+                    ],
+                )
+                # Create a mask indicating which pixels survived the cleaning
+                clean_mask = np.zeros_like(image, dtype=bool)
+                clean_mask[clean] = True
+
+                if np.sum(clean_mask) == 0:
+                    continue
+
+                # Apply the mask to relevant data arrays
+                clean_peak_time = peak_time[clean_mask]
+                clean_image = image[clean_mask]
+                clean_camgeom = camgeom[clean_mask]
+
+            elif tel_ids["MAGIC-I"] == tel_id or tel_ids["MAGIC-II"] == tel_id:
+                # Use the wrapper function to clean the image
+                clean_mask, image, peak_time = clean_image_with_modified_thresholds(
+                    event_image=image,
+                    event_pulse_time=peak_time,
+                    unsuitable_mask=unsuitable_mask,
+                    magic_clean=magic_clean,  # Pass the magic_clean instance here
+                    config_clean_magic=config_clean_magic,
+                    cmf=cmf,
+                )
+
+                clean_camgeom = camgeom[clean_mask]
```

Comment:
those 3 lines are exactly like lines 488-490, therefore can be extracted out of the loop.
the next 2 lines  are different then lines 484-485 but play the same role - removing of empty events, so this part can be also extracted out of the loop with any of the two approaches